.tekton/tasks/check-app-health.yaml

```diff
@@ -0,0 +1,89 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: argocd-app-health-check
+spec:
+  description: >-
+    This task is for checking the health of the staging argocd app
+  workspaces:
+   - name: argocd-config
+  steps:
+    - name: check-application-status
+      image: quay.io/devtools_gitops/test_image:4.0.5
+      workingDir: $(workspaces.argocd-config.path)
+      script: |
+        #!/usr/bin/env sh
+        set -u -o pipefail
```

Comment:
You should set the whole gammut of
```
set -o errexit
set -o nounset
set -o pipefail
```

(also applies to other sh scripts in other yamls)